src/librustc_mir/util/liveness.rs

```diff
@@ -220,6 +220,81 @@ impl LivenessResult {
     }
 }
 
+#[derive(Eq, PartialEq, Clone)]
+pub enum DefUse {
+    Def,
+    Use,
+}
+
+pub fn categorize<'tcx>(context: PlaceContext<'tcx>, mode: LivenessMode) -> Option<DefUse> {
+    match context {
+        ///////////////////////////////////////////////////////////////////////////
+        // DEFS
+
+        PlaceContext::Store |
+
+        // This is potentially both a def and a use...
+        PlaceContext::AsmOutput |
+
+        // We let Call define the result in both the success and
+        // unwind cases. This is not really correct, however it
+        // does not seem to be observable due to the way that we
+        // generate MIR. See the test case
+        // `mir-opt/nll/liveness-call-subtlety.rs`. To do things
+        // properly, we would apply the def in call only to the
+        // input from the success path and not the unwind
+        // path. -nmatsakis
```

Comment:
`call` leaves garbage in the destination on the unwind path, so the old value of the variable is still irrelevant.